### PR TITLE
[AC-5444] - Fix make targets in the new dev environment

### PR DESCRIPTION
**Pull request by yotammanor**
**Changes Introduced in [AC-5444](https://masschallenge.atlassian.net/browse/AC-5444):**

- fix `delete-vms` command such that it doesn't break badly when there's nothing to delete.
- in target `dump-db` quiet a few make commands by adding missing `@` signs.
- clean up `initial.sql` such that it has only DB configuration info, and no stale data.
- load initial_schema.sql.gz (or a configurable value) when `make run-server` is executed for the first time after a `make build`.
- `make upload-db` runs from a container, and does not rely on a locally installed aws-cli.

**Test:**
- in impact, run `make dump-db`. On development, a few commands will be echoed. On branch no more.
- in impact, run `make delete-vms;make delete-vms` (twice). On development, there will be a make error. On branch there won't be.
- run `make build;make run-server` on branch, and wait. When done, you'll see that the DB is an initial_schema as expected.
  - change the data in the db. dump the database with a specific filename. run `make build;make run-server gz_file=that_file_name.sql.gz`. when the server is finally up, see that the change persisted.
- test the new `make upload-db`:
  - run `make load-remote-db` to make sure you have the latest initial schema
  - edit `.dev.env` with AWS access and secret keys, as in [OpsWorks Test 4](https://console.aws.amazon.com/opsworks/home?region=us-east-1#/stack/67a723c8-ea0a-49ce-a7ad-45427de81248/show). You can also use your default user credentials that should exist in `~/.aws/credentials`
  - run `make upload-db`. This will build a new docker dontainer, and upload initial schema (or whatever file you specified as gz_file parameter).
  - stretch: make sure your local host doesn't have `aws-cli` configured (running `aws --version` should fail. This can be easily done by running from inside a fresh virtualenv).

### CODE REVIEW

**Comment by yotammanor** [comment]
This PR is downstream from an unmerged branch #133.

**Review by pbx — CHANGES_REQUESTED** [review]
Looks good, and I did all the manual tests successfully. 

I'd like it if we could rework `make upload-db` a bit so that it's harder to accidentally overwrite `initial_schema.sql.gz` in our shared S3 store (which happens if you omit the `db_name` variable). Perhaps use the gz_file's basename by default if `db_name` is not specified?

Even so, if the two typos are fixed I'm OK with merging.

File: Makefile [hunk 1]
```diff
@@ -97,6 +97,9 @@ target_help = \
   '\tReports any errors from the different repos.' \
   ' ' \
   'run-server - Starts the local server. Set debug=0 for supervisor output.' \
+  '\t When running the server for the first time after a build, the database' \
+  '\t sepcified by $(gz_file) will be loaded.' \
```
[the remark this answers is not in the document]
Comment (yotammanor) [reply Makefile]: fixed, thanks!

[the remark this answers is not in the document]
Comment (yotammanor) [reply Makefile]: done, and in another location as well. Thanks.

**Comment by yotammanor** [comment]
@pbx addressed your inline comments, plus I added an "Are You Sure?" Prompt to upload db, and a note on the required AWS configuration.

**Review by pbx — APPROVED** [review]

**Pull request merged**